go.mod

```diff
@@ -1,11 +1,17 @@
 module github.com/clerkinc/clerk-sdk-go
 
-go 1.16
+go 1.21
```

Comment:
🔧 I think a Go version upgrade may require a separate PR and (at least) a minor version bump, as this is a breaking change as far as I understand.